src/Alterway/Bundle/RestHalBundle/ApiResource/Resource.php

```diff
@@ -2,47 +2,38 @@
 
 namespace Alterway\Bundle\RestHalBundle\ApiResource;
 
-use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\Routing\RouterInterface;
 
 abstract class Resource implements ResourceInterface
 {
-    /**
-     * @var ProxyResource
-     */
-    protected $hal;
-
-    /**
-     * @var RouterInterface
-     */
-    protected $router;
+    protected ProxyResource $hal;
+    protected RouterInterface $router;
 
     public function __construct(RouterInterface $router)
     {
         $this->router = $router;
         $this->hal = new ProxyResource();
     }
 
-    public function addLink($rel, $route, array $routeParams = array(), array $attributes = array())
+    public function addLink($rel, $route, array $routeParams = array(), array $attributes = array()): Resource
     {
         $generatedRoute = $this->generate($route, $routeParams);
         $this->hal->addLink(
             $rel,
             $generatedRoute,
-            null,
             $attributes
         );
 
         return $this;
     }
 
-    public function addResource($rel, ResourceInterface $resource = null)
+    public function addResource($rel, ?ResourceInterface $resource = null)
     {
         $this->hal->addResource($rel, $resource->getHal());
```

Comment:
`$resource` peut être `null` ici,  on ne peut pas appeler une méthod sans vérifier que ce n'est pas le cas

Reply:
La méthode `$this-hal->addResource()` permet de passer une resource null du coup je n'ai pas à ajouter de logique de mon côté

Reply:
Le problème ici, c'est `$resource->getHal()`. Si `$resource === null` ,  ça fait une erreur.